### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by shzeshzeshze** [input-output-hk/daedalus#2575]
Wondering if improved accessibility is in the works? Or has the team settled on an accessibility standard, something like WCAG 2.0 AA or 2.1 AA? I haven't done a full manual test of Daedalus as I'm just trying it out for the first time today but from a very quick screen reader test (two very important things: using the main navigation and navigating to pages) the main icon navigation has no discernible text. When a screen reader user focuses a particular button in the navigation it just reads _button_. When a screen reader user clicks on one of these navigation links the screen reader alerts to a button being pressed but then nothing. I haven't dug around in the code yet but presumably the routing is being done via Javascript and because there is no page reload a screen reader user is unaware their button click has done anything. Ideally the user will be alerted of a page change and the focus should be programmatically handles according.

According to teh CDC, in the U.S. alone there are [61 million people who have some sort of disability](https://www.cdc.gov/ncbddd/disabilityandhealth/infographic-disability-impacts-all.html#:~:text=61%20million%20adults%20in%20the,is%20highest%20in%20the%20South.). I can imagine this number is much greater world-wide. I think this warrants investigating in order to provide an equivalent experience to all.

Thanks for your time.

**Comment by nikolaglumac** [comment]
cc @darko-mijic

**Comment by shzeshzeshze** [comment]
To remediate the SidebarCategory buttons lack of disernable text you could [add an `aria-label` to the <button> element](https://github.com/shzeshzeshze/daedalus/commit/e2633a6901a767e98c75b80e75477723a3d6acbc). Assuming `name` is a human understandable value (e.g. Wallets, Staking, Voting, etc.).

Remediating accessibility on a single page application is probably a little more involved. Here is a nice write up explaining [one method of accessible routing in React](https://timwright.org/blog/2019/03/23/accessible-routing-in-react/). 

Basically we'd want to create a "Messages" component. This can render something as simple as a visually hidden <div> with the ARIA attributes [`aria-live="polite" aria-atomic="true"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions). The messages in this case would be about page loads, so things like: "The wallets page has loaded, The staking page has loaded, The voting page has loaded." The messages should be informative, brief, and in most cases removed right after they have be inserted into the component. `aria-live="polite"` should be used in this case. It will let the screen reader finish reading whatever it may be reading before announcing our message.

Once the client is done rendering the new view we are still left with the focus on the SidebarCategory button. It would be useful, once React has rendered the new view, to move focus to the new content. Presumably a <h1> for that page. If you'd run into an issue with that particular element receiving focus, adding `tabindex="-1"` to desired focus' container should do the trick.

**Comment by disassembler** [comment]
Closing as part of issue backlog cleanup — issues older than 90 days are being retired to give the new team a clean slate. Please open a new issue if this is still relevant.
